pkg/agent/util/net_linux_test.go

```diff
@@ -0,0 +1,827 @@
+			if tc.wantErrStr == "" {
+				assert.EqualValues(t, (*net.Interface)(testNetInterface), gotInterface)
+				assert.NoError(t, gotErr)
+			} else {
+				assert.Nil(t, gotInterface)
+				assert.ErrorContains(t, gotErr, tc.wantErrStr)
+			}
+		})
+	}
+}
+
+func TestGetNSPath(t *testing.T) {
+	testPath := "test-path"
+	testNetNS := &mockNetNS{file: testPath}
+	tests := []struct {
+		name       string
+		getNSErr   error
+		wantPath   string
+		wantErrStr string
+	}{
+		{
+			name:     "Get NS Path",
+			wantPath: testPath,
+		},
+		{
+			name:       "Get NS Err",
+			getNSErr:   testInvalidErr,
+			wantErrStr: "failed to open netns",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			defer mockGetNS(testNetNS, tc.getNSErr)()
+			defer mockNetNSPath(testNetNS)()
+			defer mockNetNSClose(testNetNS)()
+			gotPath, gotErr := GetNSPath(testPath)
+			assert.Equal(t, tc.wantPath, gotPath)
+			if tc.wantErrStr == "" {
+				assert.NoError(t, gotErr)
+			} else {
+				assert.ErrorContains(t, gotErr, tc.wantErrStr)
+			}
+		})
+	}
+}
+
+func TestSetLinkUp(t *testing.T) {
+	testLink := mockLink{
+		index:        1,
+		name:         "antrea-en0",
+		hardwareAddr: testMACAddr,
+	}
+	tests := []struct {
+		name          string
+		expectedCalls func(mockNetlink *netlinktest.MockInterfaceMockRecorder)
+		wantErr       error
+		wantMac       net.HardwareAddr
+		wantIndex     int
+	}{
+		{
+			name: "Set Link Up Normal",
+			expectedCalls: func(mockNetlink *netlinktest.MockInterfaceMockRecorder) {
+				mockNetlink.LinkByName("antrea-en0").Return(testLink, nil)
+				mockNetlink.LinkSetUp(testLink).Return(nil)
+			},
+			wantMac:   testMACAddr,
+			wantIndex: 1,
+		},
+		{
+			name: "Get Link Err",
+			expectedCalls: func(mockNetlink *netlinktest.MockInterfaceMockRecorder) {
+				mockNetlink.LinkByName("antrea-en0").Return(nil, testInvalidErr)
+			},
+			wantErr:   testInvalidErr,
+			wantIndex: 0,
+		},
+		{
+			name: "Get Link Not Found",
+			expectedCalls: func(mockNetlink *netlinktest.MockInterfaceMockRecorder) {
+				mockNetlink.LinkByName("antrea-en0").Return(nil, netlink.LinkNotFoundError{})
+			},
+			wantErr: LinkNotFound{
+				fmt.Errorf("link antrea-en0 not found"),
+			},
+			wantIndex: 0,
+		},
+		{
+			name: "Setup Link Err",
+			expectedCalls: func(mockNetlink *netlinktest.MockInterfaceMockRecorder) {
+				mockNetlink.LinkByName("antrea-en0").Return(testLink, nil)
+				mockNetlink.LinkSetUp(testLink).Return(testInvalidErr)
+			},
+			wantErr:   testInvalidErr,
+			wantIndex: 0,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			defer mockUtilNetlink(ctrl, tc.expectedCalls)()
+			defer mockUtilNetlinkAttrs(testLink)()
+			gotMac, gotIndex, gotErr := SetLinkUp("antrea-en0")
+			assert.Equal(t, tc.wantMac, gotMac)
+			assert.Equal(t, tc.wantIndex, gotIndex)
+			assert.Equal(t, tc.wantErr, gotErr)
```

Comment:
Use `assert.EqualErrorf`. So do others.

Reply:
May I ask the purpose of using `assert.EqualErrorf` instead of `assert.Equal`? Because according to the definition"_EqualErrorf asserts that a function returned an error (i.e. not `nil`) and that it is equal to the provided error._", then in the case of `nil` error, this introduces more case check.

Reply:
Kept most cases, but changed `required.NoError` inspired by other PR.